lib/internal/util/iterable-weak-map.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const {
+  Set,
+  SetPrototypeAdd,
+  SetPrototypeDelete,
+  Symbol,
+  WeakMap,
+  WeakMapPrototypeGet,
+  WeakMapPrototypeDelete,
+  WeakMapPrototypeSet,
+} = primordials;
+
+// This class is modified from the example code in the WeakRefs specification:
+// https://github.com/tc39/proposal-weakrefs
+// Licensed under ECMA's MIT-style license, see:
+// https://github.com/tc39/ecma262/blob/master/LICENSE.md
+class IterableWeakMap {
+  #weakMap = new WeakMap();
+  #refSet = new Set();
+  #finalizationGroup = new global.FinalizationRegistry(cleanup);
```

Comment:
I'm not sure we have a policy around that kind of thing in core, but I think it makes more sense to use `global` for Node.js-specific globals (E.G.: `process`), and `globalThis` for ECMAScript globals.
```suggestion
  // TODO(aduh95): Add FinalizationRegistry to primordials
  #finalizationGroup = new globalThis.FinalizationRegistry(cleanup);
```